packages/protoplugin/src/ecmascript/transpile.ts

```diff
@@ -0,0 +1,103 @@
+// Copyright 2021-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import type { FileInfo } from "./generated-file.js";
+import ts from "typescript";
+import {
+  createDefaultMapFromNodeModules,
+  createSystem,
+  createVirtualCompilerHost,
+} from "@typescript/vfs";
+
+/* eslint-disable import/no-named-as-default-member */
+
+function createTranspiler(options: ts.CompilerOptions, files: FileInfo[]) {
```

Comment:
I forgot: Can you add a comment somewhere that explains 1. that we intentionally transpile with a pinned older version of TypeScript for stability, and 2. that we put @typescript/vfs in the top-level package.json in a conflicting version to avoid npm hoisting it to the top-level node_modules directory, where it would pick up the unpinned TypeScript version?